### PR TITLE
Run all tests

### PR DESCRIPTION
**Pull request by phracek**
Run all tests and report results at the end.

This commit executes all tests and report
at the end result in all of them.

Some tests have to be fixed. It failed.
E.g. `test_dev_mode app false production`.
Some tests were added like: `test_nodemon_present`, `test_npm_cache_exists` for development mode.

The Review is needed. Although the result test suite looks good.

Results:
```bash
 [PASSED] for 'app' test_run_app_application
 [PASSED] for 'app' test_s2i_usage
 [PASSED] for 'app' test_scl_usage
 [PASSED] for 'app' test_connection
 [PASSED] for 'app' test_docker_run_usage
 [PASSED] for 'app' test_scl_variables_in_dockerfile
 [PASSED] for 'app' test_npm_functionality
 [PASSED] for 'app' test_check_build_using_dockerfile
 [PASSED] for 'app' test_nodemon_removed
 [PASSED] for 'app' test_npm_cache_cleared
 [PASSED] for 'app' test_npm_tmp_cleared
 [PASSED] for 'app' kill_test_application
 [PASSED] for 'app' test_dev_mode_app_true_development
 [PASSED] for 'app' test_dev_mode_app_false_production
 [PASSED] for 'development' test_run_app_application
 [PASSED] for 'development' test_connection
 [PASSED] for 'development' test_nodemon_present
 [PASSED] for 'development' test_npm_cache_exists
 [PASSED] for 'development' kill_test_application
 [PASSED] for 'development' test_dev_mode_app_true_development
 [PASSED] for 'development' test_dev_mode_app_false_development
 [PASSED] for 'hw' test_safe_logging
 [PASSED] for 'hw' test_run_hw_application
 [PASSED] for 'hw' test_incremental_build
 [PASSED] for 'hw' test_build_express_webapp
 [PASSED] for 'hw' test_running_express_js
```

Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>

### CODE REVIEW

**Comment by phracek** [comment]
@hhorak @pkubatrh @fila43  PTAL

I have added some tests and fixed some test which failed. Especially for `DEVELOPMENT` usage.

**Comment by phracek** [comment]
[test][test-openshift]

File: test/run [hunk 1]
```diff
@@ -336,153 +402,147 @@ run_s2i_build_express_webapp() {
...
-run_express_test_suite; check_result $?
-
-# These tests only make sense for rhscl
-if [ "$OS" == "rhel7" ] || [ "$OS" == "centos7" ]; then
-  # autocleanup only enabled here as only the following tests so far use it
-  CID_FILE_DIR=$(mktemp -d)
-  ct_enable_cleanup
-
-  echo "Testing npm availability in Dockerfile"
-  ct_binary_found_from_df npm
```
Comment (fila43) [inline test/run:469]: ct_binary_found_from_df is no more useful?

Comment (phracek) [reply test/run:469]: Sorry I have missed that;) I will fix.

File: test/run [hunk 2]
```diff
@@ -235,6 +281,9 @@ test_scl_usage() {
     return 1
   fi
 }
+function test_scl_usage() {
+  scl_usage "node --version" "v$VERSION."
```
Comment (fila43) [inline test/run:285]: I didn't find chceck_result $? for test_scl_usage

**Comment by fila43** [comment]
@phracek thanks, now It looks good, LGTM

**Comment by phracek** [comment]
[test]

**Pull request merged**